### PR TITLE
LPD-3383

### PR DESCRIPTION
**Pull request by brianikim**
@andrea-ale-sbarra ,

Resent from: https://github.com/liferay-commerce/liferay-portal/pull/4795

https://liferay.atlassian.net/browse/LPD-3383

We support exclusion for validation on the Asset side. We can exclude validation when it is not a publish attempt.

cc/ @alessiorendina 

Sincerely,
Brian I. Kim

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:stop

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
@andrea-ale-sbarra , any updates on this PR?

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/asset/validator/CPDefinitionAssetEntryValidatorExclusionRule.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	@Override
+	public boolean isValidationExcluded(
+		long groupId, String className, long classPK, long classTypePK,
+		long[] assetCategoryIds, String[] assetTagNames) {
+
+		CPDefinition cpDefinition = _cpDefinitionLocalService.fetchCPDefinition(
+			classPK);
+
+		ServiceContext serviceContext =
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/asset/validator/CPDefinitionAssetEntryValidatorExclusionRule.java:36]: I'm not sure brian will merge this, I remember he discouraged the use of this thread local..

@alessiorendina can you please have a look at this too?

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/asset/validator/CPDefinitionAssetEntryValidatorExclusionRule.java:36]: I've just asked brian directly and he says it's ok in this case 👍

**Comment by andrea-ale-sbarra** [comment]
ci:test:commerce-functional

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by andrea-ale-sbarra** [comment]
too many failures I'm relaunching ci:test:commerce-functional

**Comment by andrea-ale-sbarra** [comment]
ci:test:commerce-functional

**Comment by andrea-ale-sbarra** [comment]
failures look coherent with daily failures analysis. I'll forward this PR

**Comment by andrea-ale-sbarra** [comment]
ci:forward:force

**Pull request closed without merging**